File: Python/BinaryTree/InorderTraversal.py
#
# @author
# aakash.verma
#

class InorderTraversal:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        inorder = []
        stack = []
        if root is None:
            return inorder
        curr = root
        while curr is not None or len(stack) != 0:
            while curr is not None:
                stack.append(curr)
                curr = curr.left
            curr = stack.pop()
            inorder.append(curr.val)
            curr = curr.right
        return inorder

if __name__ == '__main__':
    .
    .
    .

File: C++/BinaryTree/InorderTraversal.cpp
/**
 * @author
 * aakash.verma
 *
 */

#include <bits/stdc++.h> 
using namespace std;

class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> inorder;
        stack<TreeNode*> stack;
        if(root == NULL) {
            return inorder;
        }
        
        TreeNode *curr = root;
        while(curr != NULL || stack.size() != 0) {
            while(curr != NULL) {
                stack.push(curr);
                curr = curr->left;
            }
            curr = stack.top();
            stack.pop();
            inorder.push_back(curr->val);
            curr = curr->right;
        }
        return inorder;
    }
};

int main() {
    .
    .
    .
}